### PR TITLE
Add s390x arch support

### PR DESCRIPTION
**Pull request by rohitsakala**
Add s390x support to rancher/rancher

Part of https://github.com/rancher/rancher/issues/35501

### CODE REVIEW

**Review by luthermonson — APPROVED** [review]

**Review by luthermonson — APPROVED** [review]

File: Dockerfile.dapper [hunk 1]
```diff
@@ -20,7 +20,13 @@ ENV DOCKER_URL_amd64=https://get.docker.com/builds/Linux/x86_64/docker-1.10.3 \
...
     DOCKER_URL=DOCKER_URL_${ARCH}
 
-RUN wget -O - ${!DOCKER_URL} > /usr/bin/docker && chmod +x /usr/bin/docker
+RUN if [ "${ARCH}" == "s390x" ]; then \
+        curl -L https://download.docker.com/linux/static/stable/s390x/docker-18.06.3-ce.tgz | tar xzvf - && \
+        cp docker/docker /usr/bin/ && \
+        chmod +x /usr/bin/docker; \
+    else \
+        wget -O - ${!DOCKER_URL} > /usr/bin/docker && chmod +x /usr/bin/docker; \
+    fi
```
Comment (thedadams) [inline Dockerfile.dapper:29]: Can this be changed to use the same pattern as above?

File: .drone.yml [hunk 2]
```diff
@@ -150,8 +226,10 @@ steps:
       platforms:
         - linux/amd64
         - linux/arm64
+        - linux/s390x
       target: "rancher/machine:${DRONE_TAG}"
       template: "rancher/machine:${DRONE_TAG}-ARCH"
+      ignore_missing: true
```
Comment (thedadams) [inline .drone.yml:232]: Ignoring something that is missing seems like we are ignoring a problem. Is this necessary?

**Review by thedadams — CHANGES_REQUESTED** [review]

File: .drone.yml (hunk 2 above)
Comment (rohitsakala) [reply .drone.yml:232]: All steps in s390x pipeline https://github.com/rancher/machine/pull/154/files#diff-b54b39f1afced2465e1f3641db9d5bbf4f3a7fcf890996dfedd3c197bcb7f8c7R153 has `failure:ignore`. So if s390x pipeline fails, the relevant docker image for s390x is not found. But the manifest pipeline will not fail due to `ignore_missing: true` thus not blocking rancher releases due to problems in s390x pipeline.....

File: Dockerfile.dapper (hunk 1 above)
Comment (rohitsakala) [reply Dockerfile.dapper:29]: Unfortunately no, docker binary for s390x is inside a compressed file and for s390x, tar is necessary.

We can get everything into the same pattern, if we can update the docker from 1.10.3 to 18.06.3. But the problem here is for arm, docker binary is being used from a fork repo of rancher https://github.com/rancher/machine/blob/master/Dockerfile.dapper#L19. So not sure if we can use docker binary from https://download.docker.com/linux/static/stable/.

File: .drone.yml (hunk 2 above)
Comment (luthermonson) [reply .drone.yml:232]: machine releases independently and it's process is much simpler and to me personally im comfortable dropping the failure ignore and the ignore_missing as i think this repo has a little more wiggle room than rancher/rancher

Comment (rohitsakala) [reply .drone.yml]: Removed and updated the PR.

**Review by thedadams — DISMISSED** [review]

**Review by luthermonson — APPROVED** [review]

**Review by thedadams — APPROVED** [review]

**Pull request merged**